lib/actions/api.js

```diff
@@ -80,7 +80,9 @@ function getOtpFetchOptions(state, includeToken = false) {
 
   const isOtpServerSameAsMiddleware = apiBaseUrl === api.host
   if (isOtpServerSameAsMiddleware) {
-    if (includeToken && state.user) {
+    // Use access token only if the user has completed entire account setup
+    // (Otherwise every request returns an error)
+    if (includeToken && state.user?.loggedInUser?.hasConsentedToTerms) {
       const { accessToken, loggedInUser } = state.user
       if (accessToken && loggedInUser) {
```

Comment:
Another way to avoid redundant checks is to say`if (accessToken && loggedInUser?.hasConsentedToTerms)` on this line.

Reply:
You mean instead of the change on line 85?

Reply:
Yes.

Reply:
I see, much cleaner! Thank you